gooey-macros/src/animation.rs

```diff
@@ -0,0 +1,65 @@
+use attribute_derive::FromAttr;
+use manyhow::bail;
+use quote::ToTokens;
+use syn::{Field, ItemStruct};
+
+use crate::*;
+
+#[derive(FromAttr)]
+#[attribute(ident = interpolate)]
+struct FieldAttributes {
+    skip: bool,
+}
+
+pub fn linear_interpolate(
+    ItemStruct {
```

Comment:
This should also be supported for enums too. It's a main use case I want for this, as it allows hooking data types up to a slider directly.

Other traits that are required to hook a type up to a slider: `PercentBetween` and `Ranged`.